Errors creating/getting blank wiki pages

The source of the issue is that the assertion that data.revisionBy?.data is not null can fail here for wiki pages without a revision history: https://github.com/reddit/devvit/blob/9c982244e1b492eb9a7d015d63008c093d97c083/packages/public-api/src/apis/reddit/models/WikiPage.ts#L101

It affects both the `createWikiPage()` and `getWikiPage()` methods. The `revision_date`, `revision_id`, and `revision_id` properties returned by `/r/subreddit/wiki/somewikipage.json` can be null if it was created with no content and no other edits have been made since.

![image](https://github.com/user-attachments/assets/8ac01792-4231-4bca-9c2e-791727bfbed5)
![image](https://github.com/user-attachments/assets/b98d7c38-8c31-4b77-9cca-4724da658a25)

---

Reproduction:

1. Set up a Devvit button, form, etc

2. Run the following:  
```ts
await context.reddit.createWikiPage({
    content: "", // Create Wiki Page with empty content.
    page: "somenewwikipage", // Any valid wiki page as long as it hasn't been created yet
    subredditName: context.subredditName,
});
```

3. That will produce the following error:  
```
[ERROR] Nov 5 17:59:56 Error: Wiki page author details are missing
    at assertNonNull (main.js:11588:11)
    at new _WikiPage (main.js:27558:5)
    at _WikiPage.getPage (main.js:27656:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.formHandler4 [as onSubmit] (main.js:28242:29)
    at async Devvit.handleUIEvent [as HandleUIEvent] (main.js:17588:5)
    at async executeWithSourceMap (/srv/index.cjs:136468:12)
    at async /srv/index.cjs:122669:27
```

4. If you then visit the URL for the wiki page or check the moderation log, you can see that the page was in fact created.
![image](https://github.com/user-attachments/assets/03335e84-974c-425d-80a7-32e6d90a9777)
![image](https://github.com/user-attachments/assets/3381b71f-c432-4422-9eb2-8b63d3189337)

5. You can then use the same wiki page to also replicate the getWikiPage() error:  
```ts
await context.reddit.getWikiPage(context.subredditName, "somenewwikipage");
```

6. That will produce the same error as in point 3.

---

It's worth noting that Devvit is not the only way to create these types of blank wiki pages. A user can also create this type of empty wiki page by navigating to `https://old.reddit.com/r/subreddit/wiki/create/somenewwikipage` and hitting submit without entering anything into the text field.

The issue is present in 0.11.2 and goes back at least as far as 0.10.16.

--- a/packages/public-api/src/apis/reddit/models/WikiPage.ts
+++ b/packages/public-api/src/apis/reddit/models/WikiPage.ts
@@ -75,7 +75,7 @@ export class WikiPage {
   #revisionId: string;
   #revisionDate: Date;
   #revisionReason: string;
-  #revisionAuthor: User;
+  #revisionAuthor: User | undefined;
 
   #metadata: Metadata | undefined;
 
@@ -97,9 +97,9 @@ export class WikiPage {
     this.#revisionId = data.revisionId;
     this.#revisionDate = new Date(data.revisionDate * 1000); // data.revisionDate is represented in seconds, so multiply by 1000 to get milliseconds
     this.#revisionReason = data.reason ?? '';
-
-    assertNonNull(data.revisionBy?.data, 'Wiki page author details are missing');
-    this.#revisionAuthor = new User(data.revisionBy.data, metadata);
+    this.#revisionAuthor = data.revisionBy?.data
+      ? new User(data.revisionBy.data, metadata)
+      : undefined;
 
     this.#metadata = metadata;
   }
@@ -140,7 +140,7 @@ export class WikiPage {
   }
 
   /** The author of this revision. */
-  get revisionAuthor(): User {
+  get revisionAuthor(): User | undefined {
     return this.#revisionAuthor;
   }
 
@@ -154,7 +154,7 @@ export class WikiPage {
     | 'revisionDate'
     | 'revisionReason'
   > & {
-    revisionAuthor: ReturnType<User['toJSON']>;
+    revisionAuthor: ReturnType<User['toJSON']> | undefined;
   } {
     return {
       name: this.#name,
@@ -164,7 +164,7 @@ export class WikiPage {
       revisionId: this.#revisionId,
       revisionDate: this.#revisionDate,
       revisionReason: this.#revisionReason,
-      revisionAuthor: this.#revisionAuthor.toJSON(),
+      revisionAuthor: this.#revisionAuthor?.toJSON(),
     };
   }
 

